src/components/ProjectSettingsTabs/GithubCommitQueue/getFormSchema.ts

```diff
@@ -0,0 +1,350 @@
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  fields: {},
+  schema: {
+    definitions: {
+      aliasArray: {
+        type: "array" as "array",
+        items: {
+          type: "object" as "object",
+          properties: {
+            id: {
+              type: "string" as "string",
+            },
+            alias: {
+              type: "string" as "string",
+            },
+            variant: {
+              type: "string" as "string",
+              default: "",
+            },
+            variantTags: {
+              type: "array" as "array",
+              title: "Variant Tags",
+              items: {
+                type: "string" as "string",
+                title: "Variant Tag",
+                default: "",
+              },
+            },
+            task: {
+              type: "string" as "string",
+              default: "",
+            },
+            taskTags: {
+              type: "array" as "array",
+              items: {
+                type: "string" as "string",
+                title: "Task Tag",
+                default: "",
+              },
+            },
+          },
+        },
+      },
+    },
+    type: "object" as "object",
+    properties: {
+      github: {
+        type: "object" as "object",
+        title: "GitHub",
+        properties: {
+          gitHubWebhooksEnabled: {
+            type: "null",
+            title: "GitHub Webhooks",
+            description: `GitHub webhooks ${
+              gitHubWebhooksEnabled ? "are" : "aren't"
+            } enabled.`,
+          },
+          prTestingEnabledTitle: {
+            type: "null",
+            title: "GitHub Pull Request Testing",
+            ...(useRepoSettings === undefined && {
+              description:
+                "If enabled, then untracked branches will also use the file patterns defined here for PR testing.",
+            }),
+          },
+          prTestingEnabled: {
+            type: ["boolean", "null"],
+            oneOf: radioBoxOptions(
+              ["Enabled", "Disabled"],
+              repoData?.github?.prTestingEnabled
+            ),
+          },
+          prTesting: {
+            type: "object" as "object",
+            title: "GitHub Patch Definitions",
+            description:
+              "For patches created from GitHub pull requests, Evergreen will find variants that match each variant regex, and schedule tasks that match the corresponding task regex. All regular expressions must be valid Golang regular expressions.",
+            ...overrideRadioBox(
+              "githubPrAliases",
+              [
+                "Override Repo Patch Definition",
+                "Default to Repo Patch Definition",
+              ],
+              {
+                $ref: "#/definitions/aliasArray",
+              }
+            ),
+          },
+          githubChecksEnabledTitle: {
+            type: "null",
+            title: "GitHub Commit Checks",
```

Comment:
Makes sense! To clarify for a project that is attached to a repo: should we only provide this warning if neither the project nor the repo has aliases defined?

Reply:
If the project chooses "Override Repo Configuration" then it doesn't matter if the repo has aliases defined so we can still show the message.

Reply:
Actually also: There is now a [VersionConfigEnabled boolean](https://github.com/evergreen-ci/evergreen/blob/8a425038834326c212d65289e0c9e80e48d07e7e/rest/model/project.go#L466) (Malik created https://jira.mongodb.org/browse/EVG-16185 to add this). It would be awesome if this showed as an _error_ if version control is not enabled, and a warning if it is. In which case maybe this validation should be tacked onto that ticket.

Reply:
EVG-16215 can handle validation for both cases.